PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/SplitGraphsTest.java

```diff
@@ -0,0 +1,320 @@
+package nl.tudelft.pl2016gr2.core.algorithms;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import nl.tudelft.pl2016gr2.model.Node;
+import nl.tudelft.pl2016gr2.model.OriginalGraph;
+import nl.tudelft.pl2016gr2.thirdparty.testing.utility.AccessPrivate;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.function.Consumer;
+
+/**
+ * Tests the {@link SplitGraphs} class.
+ *
+ * @author Wouter Smit
+ */
+public class SplitGraphsTest {
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  private OriginalGraph mockedGraph;
+  private ArrayList<String> genomeSet;
+  private SplitGraphs defInstance;
+  private Collection<Integer> nodeIDs;
+
+  /**
+   * Sets up mocked dependencies.
+   */
+  @Before
+  public void setup() {
+    genomeSet = new ArrayList<>();
+    for (int i = 0; i < 5; i++) {
+      genomeSet.add("testGen0" + i);
+    }
+    mockedGraph = mock(OriginalGraph.class);
+    when(mockedGraph.getGenoms()).thenReturn(genomeSet);
+    defInstance = new SplitGraphs(mockedGraph);
+    setupIterator();
+  }
+
+  private void setupIterator() {
+    nodeIDs = new ArrayList<>();
+    nodeIDs.add(0);
+    nodeIDs.add(1);
+    nodeIDs.add(3);
+
+    @SuppressWarnings("unchecked") // Necessary when mocking generic types
+        Iterator<Node> mockedIterator = (Iterator<Node>) mock(Iterator.class);
```

Comment:
@FarisElghlan IntelliJ complained if I didn't cast it and some source on the Internet confirmed that when mocking generic classes, you must cast them, but I will have a look